src/lib/statistics/table_statistics.cpp

```diff
@@ -3,16 +3,37 @@
 #include <numeric>
 #include <thread>
 
-#include "attribute_statistics.hpp"
 #include "resolve_type.hpp"
 #include "statistics/statistics_objects/abstract_histogram.hpp"
 #include "storage/table.hpp"
 #include "utils/assert.hpp"
 
 namespace opossum {
 
+template <typename ColumnDataType>
+void TableStatistics::_add_statistics_from_histogram(
+    const std::shared_ptr<AttributeStatistics<ColumnDataType>> attribute_statistics,
+    const std::shared_ptr<EqualDistinctCountHistogram<ColumnDataType>> histogram, const Table& table) {
+  if (histogram) {
+    attribute_statistics->set_statistics_object(histogram);
+
+    // Use the insight that the histogram will only contain non-null values to generate the NullValueRatio
+    // property
+    const auto null_value_ratio =
+        table.row_count() == 0
+            ? 0.0f
+            : 1.0f - (static_cast<float>(histogram->total_count()) / static_cast<float>(table.row_count()));
+    attribute_statistics->set_statistics_object(std::make_shared<NullValueRatioStatistics>(null_value_ratio));
+  } else {
+    // Failure to generate a histogram currently only stems from all-null segments.
```

Comment:
What do you mean be "currently"? Rather "at this point"?

Reply:
This code wasn't written by us.

Reply:
Can you elaborate? Did you move it or copy it?

Reply:
We moved it.